src/region.rs

```diff
@@ -99,3 +101,56 @@ impl Region {
             .ok_or_else(|| Error::custom(format!("unsupported region {v}")))
     }
 }
+
+impl slog::Value for Region {
+    fn serialize(
+        &self,
+        _record: &slog::Record,
+        key: slog::Key,
+        serializer: &mut dyn slog::Serializer,
+    ) -> slog::Result {
+        serializer.emit_str(key, &self.to_string())
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct RegionParams {
+    pub gain: u64,
+    pub region: Region,
+    pub params: Vec<BlockchainRegionParamV1>,
+}
+
+impl TryFrom<GatewayRegionParamsStreamedRespV1> for RegionParams {
+    type Error = Error;
+    fn try_from(value: GatewayRegionParamsStreamedRespV1) -> Result<Self> {
+        let region = Region::from_i32(value.region)?;
+        let params = if let Some(params) = value.params {
+            params.region_params
+        } else {
+            return Err(RegionError::no_region_params());
+        };
+        Ok(Self {
+            gain: value.gain,
+            params,
+            region,
+        })
+    }
+}
+
+impl RegionParams {
+    pub fn max_eirp(&self) -> Option<u32> {
```

Comment:
A comment on units would be helpful. Deci-dBm?

Reply:
Or maybe even a fixed-point type? I'd prefer to reduce the risk of someone down the line not taking into account a scaling factor.

Reply:
Added in [63f3de5](https://github.com/helium/gateway-rs/pull/262/commits/63f3de5bcdcd9e7cfdf0856edae28eabc98f2487)